.github/workflows/swift-toolchain.yml

```diff
@@ -762,25 +762,49 @@ jobs:
       - name: Build early swift-driver
         run: |
           $env:SWIFTCI_USE_LOCAL_DEPS=1
+          if ("${{ matrix.os }}" -eq "Windows") {
+            $LinkerFlags = @("-Xlinker", "${env:SDKROOT}/usr/lib/swift/windows/x86_64/swiftCore.lib")
+          } else {
+            $LinkerFlags = @()
+          }
 
           swift build `
             --configuration release `
             --package-path ${{ github.workspace }}/SourceCache/swift-driver `
             --build-path ${{ github.workspace }}/BinaryCache/swift-driver `
-            -Xlinker "${env:SDKROOT}/usr/lib/swift/windows/x86_64/swiftCore.lib"
+            @LinkerFlags
+
+      - name: Copy binaries
+        run: |
+          if ("${{ matrix.os }}" -eq "Windows") {
+              $ExeSuffix = ".exe"
+              if ("${{ matrix.arch }}" -eq "arm64") {
+                $Cpu = "aarch64"
+              } else {
+                $Cpu = "x86_64"
+              }
+              $SourceBinDir = Join-Path "${{ github.workspace }}" "BinaryCache" "swift-driver" "${Cpu}-unknown-windows-msvc" "release"
+          } else {
+              $ExeSuffix = ""
+              $Cpu = "${{ matrix.arch }}"
+              $SourceBinDir = Join-Path "${{ github.workspace }}" "BinaryCache" "swift-driver" "${Cpu}-apple-macosx" "release"
+          }
+          $InstallBinDir = Join-Path "${{ github.workspace }}" "BuildRoot" "bin"
 
-      - id: export-binary-paths
-        run: |
-          $Suffix = if ("${{ matrix.os }}" -eq "Windows") { ".exe" } else { "" }
-          echo "swift_driver=${{ github.workspace }}/BinaryCache/swift-driver/x86_64-unknown-windows-msvc/release/swift-driver${Suffix}" >> $env:GITHUB_OUTPUT
-          echo "swift_help=${{ github.workspace }}/BinaryCache/swift-driver/x86_64-unknown-windows-msvc/release/swift-help${Suffix}" >> $env:GITHUB_OUTPUT
+          # Create the target folder.
```

Comment:
Why do we need to do this?

Reply:
The `gha-upload-tar-artifact` action only takes a single path (file or directory). The logic used by the `upload-artifact` and `download-artifact` actions to handle multiple paths and globbing is very hard to replicate.